mlx_graphs/data/heteroData.py

```diff
@@ -0,0 +1,220 @@
+                dimension while using batching.
+
+        Returns:
+            The dimension where concatenation will happen when batching.
+        """
+        if "index" in key:
+            return 1
+        return 0
+
+    def __inc__(self, key: str, type_key: str, *args, **kwargs) -> Union[int, None]:
+        """
+        This method can be overriden when batching is used with custom attributes.
+        Given the name of a custom attribute `key` and its type `type_key`, returns an
+        integer indicating the incrementing value to apply to the elements
+        of the attribute.
+
+        By default, all attribute names containing "index" will be incremented based on
+        the number of nodes of the corresponding node type in previous batches to avoid
+        duplicate nodes in the index, e.g. `edge_index_dict`. Other attributes are not
+        incremented and keep their original values, e.g. node features.
+        If incrementation is not used, the return value should be set to `None`.
+
+        Args:
+            key: Name of the attribute on which change the default incrementation
+                behavior while using batching.
+            type_key: Type of the attribute (e.g., node type or edge type).
+
+        Returns:
+            Incrementing value for the given attribute or None.
+        """
+        if "index" in key:
+            return self.num_nodes_dict[type_key]
+        return None
+
+    def has_isolated_nodes(self, node_type: str) -> bool:
+        """Returns a boolean of whether the graph has isolated nodes of the given type
+        (i.e., nodes that don't have a link to any other nodes)"""
+        num_nodes = self.num_nodes_dict[node_type]
+        for edge_type, edge_index in self.edge_index_dict.items():
+            src_node_type, _, dst_node_type = edge_type
+            if src_node_type == node_type or dst_node_type == node_type:
+                if has_isolated_nodes(edge_index, num_nodes):
+                    return True
+        return False
+
+    def has_self_loops(self, edge_type: str) -> bool:
+        """
+        Returns a boolean of whether the graph
+        contains self loops for the given edge type.
+        """
+        return has_self_loops(self.edge_index_dict[edge_type])
+
+    def is_undirected(self, edge_type: str) -> bool:
+        """
+        Returns a boolean of whether the
+        given edge type is undirected or not.
+        """
+        edge_index = self.edge_index_dict[edge_type]
+        edge_features = (
+            self.edge_features_dict.get(edge_type, None)
+            if self.edge_features_dict
+            else None
+        )
+        return is_undirected(edge_index, edge_features)
+
+    def is_directed(self, edge_type: str) -> bool:
+        """
+        Returns a boolean of whether the given
+        edge type is directed or not.
+        """
+        return not self.is_undirected(edge_type)
```

Comment:
Shall these methods exist also for the whole graph? E.g. `is_undirected()` will return `True` if, for all the `edge_type`, the graph is undirected.